site/src/pages/WorkspacePage/WorkspacePage.tsx

```diff
@@ -42,46 +43,50 @@ export const WorkspacePage: FC = () => {
   }, [username, quotaSend])
 
   return (
-    <ChooseOne>
-      <Cond condition={workspaceState.matches("error")}>
-        <div className={styles.error}>
-          {Boolean(getWorkspaceError) && (
-            <AlertBanner severity="error" error={getWorkspaceError} />
-          )}
-          {Boolean(getTemplateWarning) && (
-            <AlertBanner severity="error" error={getTemplateWarning} />
-          )}
-          {Boolean(getTemplateParametersWarning) && (
-            <AlertBanner
-              severity="error"
-              error={getTemplateParametersWarning}
-            />
-          )}
-          {Boolean(checkPermissionsError) && (
-            <AlertBanner severity="error" error={checkPermissionsError} />
-          )}
-          {Boolean(getQuotaError) && (
-            <AlertBanner severity="error" error={getQuotaError} />
-          )}
-        </div>
-      </Cond>
-      <Cond
-        condition={
-          Boolean(workspace) &&
-          workspaceState.matches("ready") &&
-          quotaState.matches("success")
-        }
-      >
-        <WorkspaceReadyPage
-          workspaceState={workspaceState}
-          quotaState={quotaState}
-          workspaceSend={workspaceSend}
-        />
-      </Cond>
-      <Cond>
-        <Loader />
-      </Cond>
-    </ChooseOne>
+    <RequirePermission
+      isFeatureVisible={getWorkspaceError?.response?.status !== 404}
+    >
```

Comment:
`getWorkspaceError` may be undefined for a short time; however, we will always be in the loader during that period because we will be waiting on the result of `getWorkspace`. So the user should never mistakenly see a flash of content before they are redirected.

I agree this page is a bit awkwardly structured logic-wise - I was just trying to avoid carving it up right now.

What do you think would make this component clearer? Would a simple rename work or would you rather we have a `useEffect` that depends upon `getWorkspaceError` and navigates to `/workspaces` conditionally? Or maybe there's something else you had in mind?

Reply:
Sounds good to me!